### ISSUE TITLE
Debugging with Godot 4.x -> Need to prevent using Godot 3 mono run configuration

### ISSUE THREAD

**Issue by QuintillusCFC** [JetBrains/godot-support#142]
I initially noticed this on a project I'm part of, but was able to reproduce it with the [Dodge the Creeps](https://github.com/van800/godot-demo-projects/tree/net6/mono/dodge_the_creeps) demo.

0. Clone the project, and switch to the `net6` branch
1. Open the project in Rider.  Tested with both Rider 2024.1 and the latest version of the plugin, and Rider 2023.1.6 and its latest-supported version (same results).  Build the project.
2. Configure a Godot "Start and Debug" configuration.  Have set it up like so, so it starts Dodge the Creeps directly:
![image](https://github.com/JetBrains/godot-support/assets/5287977/5071161f-0b83-4735-b061-225a6cd390e3)
3. Set a breakpoint.  I set it in the NewGame() method in Main.cs
4. Start the debugger

Expected: The breakpoint is hit when I start a new game
Actual: A crossed-out circle is shown next to the line, and when I hover next to it, it reads:

```
Line 31 in Method 'NewGame'  Didn't find the associated module for the breakpoint C:\Development\Godot\godot-demo-projects\mono\dodge_the_creeps\projects\MainProject\Main.cs:31,3
```

I can go to Run -> Attach to Process, choose the Godot process, and attach, to have the breakpoints work (both in my project and in the sample project).  But for some reason it isn't happening automatically the way it does with Godot 3.5.x.

Is there something that needs to be configured differently in 4.x than 3.5.x, or is there a missing link with for getting the debugger to start automatically?

**Comment by van800** [comment]
That run configuration with Godot icon is only applicable for Godot 3.x, please use .NET Executable run configuration for Godot 4.x.

If it causes massive misunderstanding, 
I can likely find a way to hide or make invalid the Godot 3.x run configuration, when the project is Godot 4.

**Comment by van800** [comment]
Or I can change the icon to be Godot with "3" inside.

**Comment by QuintillusCFC** [comment]
> That run configuration with Godot icon is only applicable for Godot 3.x, please use .NET Executable run configuration for Godot 4.x.
> 
> If it causes massive misunderstanding, I can likely find a way to hide or make invalid the Godot 3.x run configuration, when the project is Godot 4.

Ah, yes, it works right away with the same setup, but as a .NET Executable!  Thank you!

Maybe the readme should have a note about the different configuration type for Godot 3 versus 4?  I tried to set it up the way I was used to from Godot 3.  The icon having a "3" would have been a hint that it wasn't the right type of configuration, but then it would still be necessary for people migrating to Godot 4 to figure out that .NET Executable is the correct type.  (Which makes sense now that I know it, but as someone who's only used Rider for Godot, .NET Project and Mono Remote also sound tempting)

**Comment by van800** [comment]
I have asked our designer for the icon https://youtrack.jetbrains.com/issue/RIDER-111049/Godot-icon-for-Godot-3.x

**Comment by van800** [comment]
fixed in Godot Support 2024.1.161

**Comment by Nosrick** [comment]
This fixed the issue I was having, as reported in the Godot Discord.
Thank you so much!
